sdk/src/utils.ts

```diff
@@ -271,3 +271,22 @@ export const getDeploymentData = async (
     throw new Error(`${contractName}.json or ${contractName}.wasm not found`)
   }
 }
+
+export const DENOMINATOR = BigInt(Math.pow(10, 12))
+
+export const calculatePriceImpact = (
```

Comment:
This code is from prehistory of Invarinat. Wrtie this function in good style. Here's formula:
![image](https://github.com/invariant-labs/protocol-a0/assets/43304803/288d7dc1-a409-4b4d-b59a-6bd67346b536)